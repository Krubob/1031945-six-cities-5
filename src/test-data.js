export const testInitialState = {
  DATA: {
    cities: [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`],
    offers: [],
    activeOffer: ``,
    offer: {},
    reviews: [],
    nearOffers: [],
    favoriteOffers: [],
    changedFavoriteOffer: {},
    activeCity: `Paris`,
    activeSorting: `Popular`,
    isLoading: false,
    error: null,
  },
  USER: {
    authStatus: `NO_AUTH`,
    authData: {},
    isLoading: false,
    error: null,
  },
};
