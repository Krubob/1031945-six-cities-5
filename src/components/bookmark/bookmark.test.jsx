import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";
import Bookmark from "./bookmark";
import {testInitialState} from "../../test-data";

const mockStore = configureMockStore();
const store = mockStore(testInitialState);

describe(`Bookmark snapshot`, () => {
  it(`Should Bookmark render correctly when isUserAuthorized={false} and isFavorite={false}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Bookmark
                offerId = {`1`}
                isFavorite = {false}
                onChangeFavoriteOfferStatusAction={()=>{}}
                className={`place-card`}
                bookmarkType={`PLACE_CARD_BOOKMARK`}
                isUserAuthorized = {false}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Bookmark render correctly when isUserAuthorized={false} and isFavorite={true}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Bookmark
                offerId = {`1`}
                isFavorite = {true}
                onChangeFavoriteOfferStatusAction={()=>{}}
                className={`place-card`}
                bookmarkType={`PLACE_CARD_BOOKMARK`}
                isUserAuthorized = {false}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Bookmark render correctly when isUserAuthorized={true} and isFavorite={false}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Bookmark
                offerId = {`1`}
                isFavorite = {false}
                onChangeFavoriteOfferStatusAction={()=>{}}
                className={`place-card`}
                bookmarkType={`PLACE_CARD_BOOKMARK`}
                isUserAuthorized = {true}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Bookmark render correctly when isUserAuthorized={true} and isFavorite={true}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Bookmark
                offerId = {`1`}
                isFavorite = {true}
                onChangeFavoriteOfferStatusAction={()=>{}}
                className={`place-card`}
                bookmarkType={`PLACE_CARD_BOOKMARK`}
                isUserAuthorized = {true}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
