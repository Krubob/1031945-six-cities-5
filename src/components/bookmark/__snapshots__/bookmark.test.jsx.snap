// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark snapshot Should Bookmark render correctly when isUserAuthorized={false} and isFavorite={false} 1`] = `
<button
  className="place-card__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Bookmark snapshot Should Bookmark render correctly when isUserAuthorized={false} and isFavorite={true} 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`Bookmark snapshot Should Bookmark render correctly when isUserAuthorized={true} and isFavorite={false} 1`] = `
<button
  className="place-card__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Bookmark snapshot Should Bookmark render correctly when isUserAuthorized={true} and isFavorite={true} 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
